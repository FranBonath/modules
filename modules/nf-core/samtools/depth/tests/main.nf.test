nextflow_process {

    name "Test Process SAMTOOLS_DEPTH"
    script "../main.nf"
    process "SAMTOOLS_DEPTH"

    tag "modules"
    tag "modules_nfcore"
    tag "samtools"
    tag "samtools/depth"

    test("human - no_intervals") {

        when {
            process {
                """
                input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam', checkIfExists: true)
                    ]
                input[1] = [[], []]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("human - with_intervals") {

        when {
            process {
                """
                input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam', checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'bed' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.multi_intervals.bed', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("human - no_intervals - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam', checkIfExists: true)
                    ]
                input[1] = [[], []]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("human - with_intervals - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam', checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'bed' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.multi_intervals.bed', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
