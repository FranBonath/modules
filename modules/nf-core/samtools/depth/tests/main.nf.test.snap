{
    "human - with_intervals": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,a89e2f8e1b66356c7500599fcadb8784"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7e29632bc14d18f38d63e78a08a73175"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,a89e2f8e1b66356c7500599fcadb8784"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7e29632bc14d18f38d63e78a08a73175"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-18T09:35:40.259043748"
    },
    "human - no_intervals": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,722d2c060944d6b96ae5ae00d8074657"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7e29632bc14d18f38d63e78a08a73175"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,722d2c060944d6b96ae5ae00d8074657"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7e29632bc14d18f38d63e78a08a73175"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-18T09:35:24.020831447"
    },
    "human - no_intervals - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,722d2c060944d6b96ae5ae00d8074657"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7e29632bc14d18f38d63e78a08a73175"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,722d2c060944d6b96ae5ae00d8074657"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7e29632bc14d18f38d63e78a08a73175"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-18T09:35:55.606614454"
    }
}